doc/migrations/v0.26-v0.27.md

```diff
@@ -0,0 +1,182 @@
+
+## Table of Contents
+
+- [Migrating to the new API](#migrating-to-the-new-api)
+  - [Table of Contents](#table-of-contents)
+  - [Migrating from callbacks](#migrating-from-callbacks)
+  - [Pull Streams to Streaming Iterables](#pull-streams-to-streaming-iterables)
+  - [Sample API Migrations](#sample-api-migrations)
+    - [Registering Protocol Handlers](#registering-protocol-handlers)
+    - [Dialing and Sending Data](#dialing-and-sending-data)
+    - [Checking if a peer is connected](#checking-if-a-peer-is-connected)
+    - [Pinging another peer](#pinging-another-peer)
+    - [Pubsub](#pubsub)
+      - [Getting subscribers](#getting-subscribers)
+      - [Getting subscribed topics](#getting-subscribed-topics)
+
+## Migrating from callbacks
+
+Callbacks are no longer supported in the libp2p API, as the API has now fully moved to async / await. You can see a full list of the available methods in the [API readme][api]
+
+**Before**
+```js
+libp2p.start((err) => {
+  if (err) throw err
+  console.log('libp2p started')
+})
+```
+
+**After**
+```js
+await libp2p.start()
+console.log('libp2p started')
+```
+
+## Pull Streams to Streaming Iterables
+
+The libp2p API no longer supports Pull Streams and has migrated to [Streaming Iterables][streaming_iterable]. If you would like to continue using Pull Streams in your application code, or need additional time to migrate your code base, you can leverage the conversion modules [async-iterator-to-pull-stream](https://github.com/alanshaw/async-iterator-to-pull-stream) and [pull-stream-to-async-iterator](https://github.com/alanshaw/pull-stream-to-async-iterator).
+
+For a growing list of async iterator modules, you should follow the [it-awesome repo][it_awesome].
+
+## Sample API Migrations
+
+### Registering Protocol Handlers
+
+Protocol registration is very similar to how it previously was, however, the handler now takes a single parameter containing the incoming stream and its protocol. Additionally, you can now pass an array of protocols to `.handle`, but a single string is still supported.
+
+**Before**
+```js
+const pull = require('pull-stream')
+libp2p.handle('/echo/1.0.0', (protocol, conn) => pull(conn, conn))
+```
+
+**After**
+```js
+const pipe = require('it-pipe')
+libp2p.handle(['/echo/1.0.0'], ({ protocol, stream }) => pipe(stream, stream))
+```
+
+### Dialing and Sending Data
+
+`dialProtocol` no longer takes a callback, and will now return a [Streaming Iterable][streaming_iterable] and the protocol that was successfully negotiated. The new stream can be used with async iterator modules, see [it-awesome][it_awesome], instead of pull streams.
+
+**Before**
+```js
+const pull = require('pull-stream')
+libp2p.dialProtocol(peerInfo, '/echo/1.0.0', (err, conn) => {
+  if (err) { throw err }
+
+  console.log('nodeA dialed to nodeB on protocol: /echo/1.0.0')
+
+  pull(
+    pull.values(['hey']),
+    conn,
+    pull.collect((err, data) => {
```

Comment:
`.collect` is not quite what's happening in the example below - do you want `.onEnd(op, done)` instead?

Reply:
Not sure I follow, do you mean to run `.onEnd` preceded by `.map` or similar? Replacing `collect` with `onEnd` won't return the results of the echo. The issue with collect is that `data` is an array, so using a through stream for logging the response and a sink such as `.onEnd` to initiate the pull and catch the error would be more analogous.

Reply:
nvm, I was looking at something else, you're right `onEnd` is the appropriate change.